src/confd/yang/infix-if-bridge@2024-03-06.yang

```diff
@@ -285,55 +295,75 @@ submodule infix-if-bridge {
+              key "vid";
+              description "List of VLANs associated with the Bridge.";
 
-          leaf vid {
-            type dot1q-types:vlanid;
-            description "The VLAN identifier to which this entry applies.";
-          }
+              leaf vid {
+                type dot1q-types:vlanid;
+                description "The VLAN identifier to which this entry applies.";
+              }
 
-          uses multicast;
-          uses mdb;
+              uses multicast;
+              uses multicast-filters;
 
-          leaf-list untagged {
-            type if:interface-ref;
-            description "The set of ports in the untagged set for VLAN.";
-            must "current() = ../../../../if:name
+              leaf-list untagged {
+                type if:interface-ref;
+                description "The set of ports in the untagged set for VLAN.";
+                must "current() = ../../../../if:name
                   or (/if:interfaces/if:interface[if:name = current()]/bridge-port/bridge = ../../../../if:name
                       and not(../tagged[contains(., current())]))" {
-              error-message "Port is not a member of this bridge or already set as tagged member in the same VLAN.";
-	    }
-	  }
-
-          leaf-list tagged {
-            description "The set of ports in the tagged set for VLAN.";
-            type if:interface-ref;
-            must "current() = ../../../../if:name
+                  error-message "Port is not a member of this bridge or already set as tagged member in the same VLAN.";
+                }
+              }
+
+              leaf-list tagged {
+                description "The set of ports in the tagged set for VLAN.";
+                type if:interface-ref;
+                must "current() = ../../../../if:name
                   or (/if:interfaces/if:interface[if:name = current()]/bridge-port/bridge = ../../../../if:name
                       and not(../untagged[contains(., current())]))" {
-              error-message "Port is not a member of bridge or already untagged in the same VLAN.";
+                  error-message "Port is not a member of bridge or already untagged in the same VLAN.";
+                }
+              }
             }
           }
         }
       }
     }
   }
+  deviation "/if:interfaces/if:interface/infix-if:bridge/type/ieee8021d/multicast-filters/multicast-filter/ports/port" {
+    deviate add {
+      must "/if:interfaces/if:interface[if:name = current()]/infix-if:bridge-port/bridge = ../../../../../if:name" {
+        error-message "Port is not member of bridge";
+      }
+    }
+  }
+
+  deviation "/if:interfaces/if:interface/infix-if:bridge/type/ieee8021q/vlans/vlan/multicast-filters/multicast-filter/ports/port" {
+    deviate add {
+      must "/if:interfaces/if:interface[if:name = current()]/infix-if:bridge-port/bridge = ../../../../../../../if:name" {
+        error-message "Port is not member of bridge";
+      }
+    }
+  }
```

Comment:
Is it a conscious choice to initially leave out support for adding the host to a group? I think we might need that to support certain industrial protocols. We might need to cross that _boundary_, if you will :wink: